src/utils/middlewares/azure_user_attributes_manage.ts

```diff
@@ -86,7 +101,43 @@ export const AzureUserAttributesManageMiddleware = (): IRequestMiddleware<
 
   if (subscriptionId.startsWith("MANAGE-")) {
     // MANAGE Flow
+    const errorOrMaybeAuthorizedCIDRs = await subscriptionCIDRsModel.findLastVersionByModelId(
+      [subscriptionId]
+    )();
+
+    if (E.isLeft(errorOrMaybeAuthorizedCIDRs)) {
+      winston.error(
+        `No CIDRs found for subscription|${subscriptionId}|${JSON.stringify(
+          errorOrMaybeAuthorizedCIDRs.left
+        )}`
+      );
+      return E.left<
+        IResponse<"IResponseErrorQuery">,
+        IAzureUserAttributesManage
+      >(
+        ResponseErrorQuery(
+          `Error while retrieving CIDRs tied to the provided subscription id`,
+          errorOrMaybeAuthorizedCIDRs.left
+        )
+      );
+    }
+
+    const maybeAuthorizedCIDRs = errorOrMaybeAuthorizedCIDRs.right;
+
+    if (isNone(maybeAuthorizedCIDRs)) {
+      winston.error(
+        `IAzureUserAttributesManage|CIDRs not found|${subscriptionId}`
+      );
+      return E.left<
+        IResponse<"IResponseErrorForbiddenNotAuthorized">,
+        IAzureUserAttributesManage
+      >(ResponseErrorForbiddenNotAuthorized);
+    }
+
     const authInfo: IAzureUserAttributesManage = {
+      authorizedCIDRs: maybeAuthorizedCIDRs.value.cidrs
+        ? maybeAuthorizedCIDRs.value.cidrs
+        : new Set((["0.0.0.0/0"] as unknown) as ReadonlyArray<CIDR>),
```

Comment:
How is `maybeAuthorizedCIDRs.value.cidrs` undefined? What's the meaning of this field being undefined?

Reply:
This is a typo of the previous commit, for what has been said, it is not possible that the cidrs are undefined, if you agree we will update this line of code removing undefined possibility.

Reply:
The type system should have warned you (line 140 should never be reachable. Hasn't it?